### ISSUE TITLE
Migrate email from Member to User only

### ISSUE THREAD

**Issue by nixnuex** [juntagrico/juntagrico#271]
Do make better use of Django's auth system and functions (password reset, ...), the `email` field should be moved (data migration) to the User. All referencing functions (auth, account setup, ...) have to be checked and adapted.

This is necessary because there was no introduction of a custom user model at the start of the project (or it was simply not available). There is no way to switch to a custom User model after the very first initial migration and the auth functions expect the email to be there.

This is gating #174.

**Comment by motu55** [comment]
it has to be discussed, if we add a property to the member which basically retruns user.email.
I would advise strongly against it since this would leave the request.user.member.email calls valid which can in the worst case lead to three calls to the database
on the other hand it would be nice to have the email field in the the member admin. if nobody thinks other wise i would suggest that we add the email only to the member admin if we can do it without a property on the member

**Comment by motu55** [comment]
ok folks
in the end the implementation was not so trivial basically there were issues how to ensure that the user email field was set and updated etc.
all the solutions where not satisfactory so we did what all responsible python developers do in such a case 
monkey pathing
since we require the email in the default password reset process the solution is tailored to that
in the pasword rset process the email is fetched using getattr and the value in EMAIL_FIELD so these were patched in the default User class
the advantage is tha we do not need any complicated migrations etc
basically all the benefits of the custom user model without the hassle to move it

**Comment by pascalfree** [comment]
I was reading a bit on the topic of custom user models.

Django does not recommend custom user models on reusable apps (which Juntagrico is). 
https://docs.djangoproject.com/en/3.2/topics/auth/customizing/#reusable-apps-and-auth-user-model
Instead the model would be on the cookiecutted user app. There we don't have migrations yet, so it would be feasible.

As juntagrico uses the email address of the members as the primary way of contacting them and requires it for many functions, the email address should be stored in Juntagrico as it is now.

So if a custom user model is created it should ideally behave like it is now doing with the monkey patch. I.e. it should reference the email field in the member module of Juntagrico.

It does not seem worth it to change anything yet.
If we however at some point want to continue this path, we should think of the  auth and user model stuff as something that the Juntagrico users control. We can suggest how to set it up, but Juntagrico should not depend on it too much.
